ext/js/pages/settings/dictionary-controller.js

```diff
@@ -113,8 +121,61 @@ class DictionaryEntry {
         this._enabledCheckbox.checked = value;
     }
 
+    /**
+     * @returns {Promise<boolean>}
+     */
+    async checkForUpdate() {
+        this._updatesAvailable.hidden = true;
+        const {isUpdatable, indexUrl, revision} = this._dictionaryInfo;
+        if (!isUpdatable || !indexUrl) { return false; }
+        const response = await fetch(indexUrl);
+
+        /** @type {unknown} */
+        const index = await readResponseJson(response);
+
+        if (!ajvSchemas.dictionaryIndex(index)) {
+            throw new Error('Invalid dictionary index');
+        }
+
+        const validIndex = /** @type {import('dictionary-data').Index} */ (index);
+
+        if (!this._compareRevisions(revision, validIndex.revision)) {
+            return false;
+        }
+
+        this._updatesAvailable.hidden = false;
+        return true;
+    }
+
     // Private
 
+    /**
+     * @param {string} current
+     * @param {string} latest
+     * @returns {boolean}
+     */
+    _compareRevisions(current, latest) {
```

Comment:
Maybe a unit test for this would be useful?

Reply:
Sure